x/pairing/keeper/cu_tracker_test.go

```diff
@@ -335,3 +349,85 @@ func TestTrackedCuPlanPriceChange(t *testing.T) {
 	balance := ts.GetBalance(providerAcc.Addr)
 	require.Equal(t, balanceBeforePay+originalPlanPrice, balance)
 }
+
+// TestFrozenProviderGetReward checks that frozen providers still get rewards.
+// providers can freeze themselves or be forcibly frozen due to unresponsiveness
+func TestFrozenProviderGetReward(t *testing.T) {
+	ts := newTester(t)
+	ts.setupForPayments(1, 1, 2)
+
+	clientAcc, _ := ts.GetAccount(common.CONSUMER, 0)
+	providerAcc, provider := ts.GetAccount(common.PROVIDER, 0)
+
+	relayPayment := sendRelay(ts, provider, clientAcc, []string{ts.spec.Index})
+	ts.relayPaymentWithoutPay(relayPayment, true)
+
+	balanceBeforePay := ts.GetBalance(providerAcc.Addr)
+
+	err := ts.Keepers.Pairing.FreezeProvider(ts.Ctx, provider, []string{ts.spec.Index}, "unresponsiveness")
+	require.Nil(t, err)
+
+	// advance month + blocksToSave + 1 to trigger the monthly payment
+	ts.AdvanceMonths(1)
+	ts.AdvanceBlocks(ts.BlocksToSave() + 1)
+
+	balance := ts.GetBalance(providerAcc.Addr)
+	planPrice := ts.plan.Price.Amount.Int64()
+	require.Equal(t, balanceBeforePay+planPrice, balance)
+}
+
+// TestTrackedCuDeletion makes sure that if there are two trackedCU entries
+// both with some CU, when the deletion occurs it deletes the older one
+func TestTrackedCuDeletion(t *testing.T) {
+	ts := newTester(t)
+	ts.setupForPayments(1, 1, 2)
+
+	clientAcc, client := ts.GetAccount(common.CONSUMER, 0)
+	_, provider := ts.GetAccount(common.PROVIDER, 0)
+	sub, _ := ts.Keepers.Subscription.GetSubscription(ts.Ctx, client)
+	fmt.Printf("sub.String(): %v\n", sub.String())
+	// send relay to track CU
+	relayPayment := sendRelay(ts, provider, clientAcc, []string{ts.spec.Index})
+	ts.relayPaymentWithoutPay(relayPayment, true)
+	oldMonthBlock := relayPayment.Relays[0].Epoch
+
+	// send relay after a month (opens a new trackedCU entry)
+	ts.AdvanceMonths(1).AdvanceEpoch()
+	relayPayment = sendRelay(ts, provider, clientAcc, []string{ts.spec.Index})
+	ts.relayPaymentWithoutPay(relayPayment, true)
+
+	// send another relay but to the old trackedCU entry
+	relayPayment = sendRelay(ts, provider, clientAcc, []string{ts.spec.Index})
+	relayPayment.Relays[0].Epoch = oldMonthBlock
+	sig, err := sigs.Sign(clientAcc.SK, *relayPayment.Relays[0])
+	relayPayment.Relays[0].Sig = sig
+	require.Nil(ts.T, err)
+	ts.relayPaymentWithoutPay(relayPayment, true)
+
+	keys := ts.Keepers.Subscription.GetAllSubTrackedCuIndices(ts.Ctx, client)
+	fmt.Printf("keys: %v\n", keys)
```

Comment:
remove the prints

Reply:
and add check for the keys

Reply:
done. see https://github.com/lavanet/lava/commit/50d68ca0d0879b8b49e5c837d82f6297b58c80dd